File: src/utils/dateFormatter.ts
export const formatNaturalDate = (dateStr: string): string => {
  if (!dateStr) return 'data não informada';
  
  // Clean up format like YYYY-MM-DD
  const parts = dateStr.split('-');
  if (parts.length === 3) {
    const year = parseInt(parts[0], 10);
    const month = parseInt(parts[1], 10) - 1;
    const day = parseInt(parts[2], 10);

    const fullMonthNames = [
      'janeiro', 'fevereiro', 'março', 'abril', 'maio', 'junho',
      'julho', 'agosto', 'setembro', 'outubro', 'novembro', 'dezembro'
    ];

    const fullDayNames = [
      'domingo', 'segunda-feira', 'terça-feira', 'quarta-feira',
      'quinta-feira', 'sexta-feira', 'sábado'
    ];

    try {
      const d = new Date(year, month, day);
      const dayOfWeek = fullDayNames[d.getDay()];
      const monthName = fullMonthNames[month] || 'mês';
      return `${dayOfWeek}, dia ${day} de ${monthName} de ${year}`;
    } catch (e) {
      // fallback
    }
  }
  return dateStr;
};

export const formatNaturalTime = (timeStr: string): string => {
  if (!timeStr) return '';
  const parts = timeStr.split(':');
  if (parts.length >= 2) {
    const hours = parseInt(parts[0], 10);
    const minutes = parseInt(parts[1], 10);
    if (minutes === 0) {
      return `${hours} horas`;
    }
    return `${hours} horas e ${minutes} minutos`;
  }
  return timeStr;
};

File: flutter/lib/utils/date_formatter.dart
class DateFormatter {
  static const List<String> fullMonths = [
    'janeiro', 'fevereiro', 'março', 'abril', 'maio', 'junho',
    'julho', 'agosto', 'setembro', 'outubro', 'novembro', 'dezembro',
  ];

  static const List<String> fullDays = [
    'domingo', 'segunda-feira', 'terça-feira', 'quarta-feira',
    'quinta-feira', 'sexta-feira', 'sábado',
  ];

  static const List<String> shortDays = ['Dom', 'Seg', 'Ter', 'Qua', 'Qui', 'Sex', 'Sáb'];

  static DateTime parseDate(String dateStr) {
    final parts = dateStr.split('-');
    return DateTime(
      int.parse(parts[0]),
      int.parse(parts[1]),
      int.parse(parts[2]),
    );
  }

  static String formatNaturalDate(String dateStr) {
    if (dateStr.isEmpty) return 'data não informada';
    try {
      final d = parseDate(dateStr);
      return '${fullDays[d.weekday % 7]}, dia ${d.day} de ${fullMonths[d.month - 1]} de ${d.year}';
    } catch (_) {
      return dateStr;
    }
  }

  static String formatNaturalTime(String timeStr) {
    if (timeStr.isEmpty) return '';
    final parts = timeStr.split(':');
    if (parts.length >= 2) {
      final hours = int.tryParse(parts[0]);
      final minutes = int.tryParse(parts[1]);
      if (hours != null && minutes != null) {
        if (minutes == 0) return '$hours horas';
        return '$hours horas e $minutes minutos';
      }
    }
    return timeStr;
  }

  static String monthShortName(DateTime d) {
    final names = ['jan', 'fev', 'mar', 'abr', 'mai', 'jun', 'jul', 'ago', 'set', 'out', 'nov', 'dez'];
    return names[d.month - 1];
  }

  static String todayStr() {
    final now = DateTime.now();
    return '${now.year.toString().padLeft(4, '0')}-${now.month.toString().padLeft(2, '0')}-${now.day.toString().padLeft(2, '0')}';
  }

  static String toDateStr(DateTime d) {
    return '${d.year.toString().padLeft(4, '0')}-${d.month.toString().padLeft(2, '0')}-${d.day.toString().padLeft(2, '0')}';
  }

  static String toTimeStr(DateTime d) {
    return '${d.hour.toString().padLeft(2, '0')}:${d.minute.toString().padLeft(2, '0')}';
  }

  static String formatShort(DateTime d) {
    final dd = d.day.toString().padLeft(2, '0');
    final mm = d.month.toString().padLeft(2, '0');
    final yy = d.year.toString().padLeft(4, '0');
    return '$dd/$mm/$yy';
  }
}